Compiler warnings when SW_ANALYSIS_STEPPER=OFF

stackwalk/src/callchecker.C: In member function ‘bool Dyninst::Stackwalker::CallChecker::isPrevInstrACall(Dyninst::Address, Dyninst::Address&)’:
    stackwalk/src/callchecker.C:47:20: error: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int’ [-Werror=format=]
       47 |    sw_printf("[%s:%u] - isPrevInstrACall on %lx\n", FILE__, __LINE__, addr);
          |                   ~^                                        ~~~~~~~~
          |                    |                                        |
          |                    unsigned int                             int
          |                   %u
    stackwalk/src/callchecker.C:52:23: error: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int’ [-Werror=format=]
       52 |       sw_printf("[%s:%u] - Address 0x%lx is not a call--unreadable\n",
          |                      ~^
          |                       |
          |                       unsigned int
          |                      %u
       53 |                 FILE__, __LINE__, addr);
          |                         ~~~~~~~~
          |                         |
          |                         int
    stackwalk/src/callchecker.C:60:23: error: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int’ [-Werror=format=]
       60 |       sw_printf("[%s:%u] - Found call encoded by %x to %lx (addr = %lx, disp = %lx)\n",
          |                      ~^
          |                       |
          |                       unsigned int
          |                      %u
       61 |                 FILE__, __LINE__, (int) buffer[0], target, addr, disp);
          |                         ~~~~~~~~
          |                         |
          |                         int
    stackwalk/src/callchecker.C:60:82: error: format ‘%lx’ expects argument of type ‘long unsigned int’, but argument 7 has type ‘int32_t’ {aka ‘int’} [-Werror=format=]
       60 |       sw_printf("[%s:%u] - Found call encoded by %x to %lx (addr = %lx, disp = %lx)\n",
          |                                                                                ~~^
          |                                                                                  |
          |                                                                                  long unsigned int
          |                                                                                %x
       61 |                 FILE__, __LINE__, (int) buffer[0], target, addr, disp);
          |                                                                  ~~~~             
          |                                                                  |
          |                                                                  int32_t {aka int}
    stackwalk/src/callchecker.C:105:26: error: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int’ [-Werror=format=]
      105 |          sw_printf("[%s:%u] - Found call of size %d encoded by: ",
          |                         ~^
          |                          |
          |                          unsigned int
          |                         %u
      106 |                    FILE__, __LINE__, size);
          |                            ~~~~~~~~
          |                            |
          |                            int
    stackwalk/src/callchecker.C:105:51: error: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘unsigned int’ [-Werror=format=]
      105 |          sw_printf("[%s:%u] - Found call of size %d encoded by: ",
          |                                                  ~^
          |                                                   |
          |                                                   int
          |                                                  %d
      106 |                    FILE__, __LINE__, size);
          |                                      ~~~~          
          |                                      |
          |                                      unsigned int
    cc1plus: all warnings being treated as errors
    make[2]: *** [stackwalk/CMakeFiles/stackwalk.dir/build.make:284: stackwalk/CMakeFiles/stackwalk.dir/src/callchecker.C.o] Error 1
    make[2]: *** Waiting for unfinished jobs....
    make[1]: *** [CMakeFiles/Makefile2:1146: stackwalk/CMakeFiles/stackwalk.dir/all] Error 2
    make[1]: *** Waiting for unfinished jobs....
    make: *** [Makefile:130: all] Error 2

--- a/stackwalk/src/callchecker.C
+++ b/stackwalk/src/callchecker.C
@@ -44,12 +44,12 @@ bool CallChecker::isPrevInstrACall(Address addr, Address & target)
    bool result;
    unsigned char buffer[max_call_length];
 
-   sw_printf("[%s:%u] - isPrevInstrACall on %lx\n", FILE__, __LINE__, addr);
+   sw_printf("[%s:%d] - isPrevInstrACall on %lx\n", FILE__, __LINE__, addr);
    Address start = addr - max_call_length;
    result = proc->readMem(buffer, start, max_call_length);
    if (!result)
    {
-      sw_printf("[%s:%u] - Address 0x%lx is not a call--unreadable\n",
+      sw_printf("[%s:%d] - Address 0x%lx is not a call--unreadable\n",
                 FILE__, __LINE__, addr);
       return false;
    }
@@ -57,7 +57,7 @@ bool CallChecker::isPrevInstrACall(Address addr, Address & target)
    if (buffer[max_call_length - 5] == 0xe8) {
       int32_t disp = *((int32_t *) (buffer+1));
       target = addr + disp;
-      sw_printf("[%s:%u] - Found call encoded by %x to %lx (addr = %lx, disp = %lx)\n",
+      sw_printf("[%s:%d] - Found call encoded by %x to %lx (addr = %lx, disp = %dx)\n",
                 FILE__, __LINE__, (int) buffer[0], target, addr, disp);
                 
       return true;
@@ -102,7 +102,7 @@ bool CallChecker::isPrevInstrACall(Address addr, Address & target)
 
       if (i + size == max_call_length)
       {
-         sw_printf("[%s:%u] - Found call of size %d encoded by: ",
+         sw_printf("[%s:%d] - Found call of size %u encoded by: ",
                    FILE__, __LINE__, size);
          for (unsigned j=i; j<i+size; j++) {
             sw_printf("%x ", buffer[j]);

